esda/shape.py

```diff
@@ -1,13 +1,18 @@
-import pygeos
-import geopandas, pandas
 import numpy
-from numba import njit, prange
+import pandas
+from .crand import njit, prange
+
 
 # -------------------- UTILITIES --------------------#
 def _cast(collection):
     """
     Cast a collection to a pygeos geometry array.
     """
+    try:
+        import pygeos, geopandas
+    except (ImportError, ModuleNotFoundError) as exception:
+        raise type(exception)("pygeos and geopandas are required for shape statistics.")
+
```

Comment:
I don't think this will work. The variable is not available outside of the `_cast` function so none of the shape functions sees it. We need to have try/except import at the top of the package.

Reply:
Yep, missed the equivalent try/except from esda/map_comparison.py will add now.